Pcap++/header/PcapFilter.h

```diff
@@ -484,82 +484,115 @@ namespace pcpp
 
 
 	/**
-	 * @class AndFilter
-	 * A class for connecting several filters into one filter with logical "and" between them. For example: if the 2 filters are: "IPv4 address =
-	 * x.x.x.x" + "TCP port dst = 80", then the new filter will be: "IPv4 address = x.x.x.x _AND_ TCP port dst = 80"<BR>
-	 * This class follows the composite design pattern<BR>
+	 * @class CompositeFilter
+	 * The base class for all filter classes composed of several other filters. This class is virtual and abstract, hence cannot be instantiated.<BR>
 	 * For deeper understanding of the filter concept please refer to PcapFilter.h
-	 * @todo add some methods: "addFilter", "removeFilter", "clearAllFilter"
 	 */
-	class AndFilter : public GeneralFilter
+	class CompositeFilter : public GeneralFilter
 	{
-	private:
+	protected:
 		std::vector<GeneralFilter*> m_FilterList;
 	public:
-
 		/**
-		 * An empty constructor for this class. Use addFilter() to add filters to the and condition
+		 * An empty constructor for this class. Use addFilter() to add filters to the composite filter.
 		 */
-		AndFilter() {}
+		CompositeFilter() = default;
 
 		/**
-		 * A constructor that gets a list of pointers to filters and creates one filter from all filters with logical "and" between them
+		 * A constructor that gets a list of pointers to filters and creates one filter from all filters
 		 * @param[in] filters The list of pointers to filters
 		 */
-		explicit AndFilter(std::vector<GeneralFilter*>& filters);
+		explicit CompositeFilter(const std::vector<GeneralFilter*>& filters);
 
 		/**
-		 * Add filter to the and condition
+		 * Add filter to the composite filter
 		 * @param[in] filter The filter to add
 		 */
 		void addFilter(GeneralFilter* filter) { m_FilterList.push_back(filter); }
```

Comment:
On the note of edge cases. Should not adding a `nullptr` to the filter be the caller's responsibility or the library's?
Since if `addFilter` is passed a `nullptr` the program burns in flames when parseToString is called.
(same issue with setFilters and the constructor from vector of pointers)

Ideally in C++11, I would resolve that issue by:
- using a `GeneralFilter const&` for `addFilter` & `removeFilter`.
- using a `std::vector<std::reference_wrapper<GeneralFilter>>` instead of `vector<GeneralFilter*>`
but those are breaking changes in the API.

Reply:
should be user's responsibility,  but I think we can throw an exception if it is a nullptr?

Reply:
I mean that works, but is there a specific reason pointers are used in places where passing nullptr is an invalid value?

IMO, a reference better describes an API that should not receive a nullptr, since the compiler would not allow it in the first place, allowing the error to be caught at compile time instead of runtime.

Perhaps a gradual effort to move to references instead of pointers for sections where nullptr value is invalid might be beneficial?

Reply:
It's just legacy code. I agree with your statement. However, we consider backward compatibility the most important. There isn't much benefit to changing to use reference by breaking API/deprecating API in this case.

Reply:
Alrighty then. As for the potential exception throw, `std::invalid_argument` or derivative of it could work, I guess?
Or just ignore the nullptr and do not throw anything?

Reply:
I guess `std::invalid_argument` make sense, please follow the way how other code throws

Reply:
Yes, we can throw an exception in this case, although I don't think it's critical because as you mentioned, the program will anyway crash if `nullptr` is being passed to the composite filter

Reply:
Alright, leaving it as it is.